src/app/AppLayout/JmxAuthForm.tsx

```diff
@@ -52,36 +51,33 @@ export const JmxAuthForm: React.FunctionComponent<JmxAuthFormProps> = (props) =>
   const [username, setUsername] = React.useState('');
   const [password, setPassword] = React.useState('');
 
-  const clear = () => {
+  const clear = React.useCallback(() => {
     setUsername('');
     setPassword('');
-  };
+  }, [setUsername, setPassword]);
 
-  const handleSave = () => {
-    context.target
-      .target()
-      .pipe(first())
-      .subscribe((target) => {
-        context.target.setCredentials(target.connectUrl, `${username}:${password}`);
-        context.target.setAuthRetry();
-        clear();
-        props.onSave(username, password);
-      });
-  };
+  const handleSave = React.useCallback(() => {
+    props.onSave(username, password).then(() => {
+      console.log('on save, setting auth retry');
+      clear();
+      context.target.setAuthRetry();
+    });
+  }, [context, context.target, clear, props.onSave, username, password]);
 
-  const handleDismiss = () => {
+  const handleDismiss = React.useCallback(() => {
     clear();
     props.onDismiss();
-  };
+  }, [clear, props.onDismiss]);
 
-  const handleKeyUp = (event: React.KeyboardEvent): void => {
+  const handleKeyUp = React.useCallback((event: React.KeyboardEvent): void => {
     if (event.keyCode === EnterKeyCode) {
```

Comment:
This line gives me `'keyCode' is deprecated` as said at https://developer.mozilla.org/en-US/docs/Web/API/KeyboardEvent/keyCode. Should we change it to something like:

```ts
if (event.key == 'Enter') { ... }
```

This is recommended as https://developer.mozilla.org/en-US/docs/Web/API/KeyboardEvent/code